app/code/community/Bolt/Boltpay/Model/ShippingAndTax.php

```diff
@@ -271,11 +271,11 @@ public function getShippingAndTaxEstimate( Mage_Sales_Model_Quote $quote, $boltO
                     )
                 );
             }
-
-        } finally {
-            $quote->setCouponCode($originalCouponCode);
+            // finally requires PHP => 5.5
+        } catch(Exception $e) {
+            // Nothing to do
         }
-
+        $quote->setCouponCode($originalCouponCode);
```

Comment:
Just as suggested in ApiController, after factoring in @daisy1754's rethrow point, this should utilize the same pattern:
```
$finallyFunction = function() { $quote->setCouponCode($originalCouponCode); };
try {
     ... 
     //we should first determine if the cart is virtual
     if($quote->isVirtual()){
          ...
          finallyFunction();
          return $response;
     }
     ...     
} catch ( Exception $e) {
     // catch-all workaround for finally not being supported in PHP 5.4
     $this->boltHelper()->notifyException($e);
     $finallyFunction();
     throw $e;
}
$finallyFunction();
return $response;
```